easy-install.py

```diff
@@ -0,0 +1,239 @@
+                    "-d",
+                ],
+                check=True,
+            )
+            
+        except Exception as e:
+            print(
+                f"{CRED} Docker Compose failed, please check the container logs{CEND}\n",
+                e,
+            )
+        print(f"\n{CGRN}Creating site: {sitename} {CEND}\n")
+        with open(os.path.join(os.path.expanduser("~"),"passwords.txt"),"w") as f:
+            f.writelines(f"Administrator:{admin_pass}\n")
+            f.writelines(f"MariaDB Root Password:{db_pass}\n")
+        try:
+            subprocess.run([
+                which("docker"),"compose",
+                "-p",project,"exec","backend",
+                "bench","new-site",sitename,
+                "--db-root-password",db_pass,
+                "--admin-password",admin_pass,
+                "--install-app","erpnext"
+                ],check=True)
+        except Exception as e:
+            print(
+                f"{CRED} Bench Site creation failed{CEND}\n",
+                e,
+            )
+    else:
+        install_docker()
+        clone_frappe_docker_repo()
+        setup_prod(project, sitename,email)  # Recursive
+
+
+def setup_dev_instance(project: str):
+    if check_repo_exists():
+        try:
+            subprocess.run(
+                [
+                    "docker",
+                    "compose",
+                    "-f",
+                    "devcontainer-example/docker-compose.yml",
+                    "--project-name",
+                    project,
+                    "up",
+                    "-d",
+                ],
+                cwd=os.path.join(os.getcwd(), "frappe_docker"),
+                check=True,
+            )
+            print(
+                f"{CYLW}Please go through the Development Documentation: https://github.com/frappe/frappe_docker/tree/main/development to fully complete the setup.{CEND}"
+            )
+        except Exception as e:
+            print(f"{CRED}Setting Up Development Environment Failed\n{CEND}", e)
+    else:
+        install_docker()
+        clone_frappe_docker_repo()
+        setup_dev_instance(project)  # Recursion on goes brrrr
+
+
+def install_docker():
+    if which("docker") is None:
+        print(f"{CGRN}Docker is not installed, Installing Docker...{CEND}")
+        if platform.system() == "Darwin" or platform.system() == "Windows":
+            print(
+                f"""{CRED}
+                This script doesn't install Docker on {"Mac" if platform.system()=="Darwin" else "Windows"}.
+
+                Please go through the Docker Installation docs for your system and run this script again{CEND}"""
+            )
+            exit(1)
+        try:
+            ps = subprocess.run(
+                ["curl", "-fsSL", "https://get.docker.com"],
+                capture_output=True,
+                check=True,
+            )
+            subprocess.run(["/bin/bash"], input=ps.stdout, capture_output=True)
```

Comment:
Run the first process as shell directly instead of manual redirection to shell?

Reply:
Tried, didn't work 😓